File: JavaScript/7js 3/scrollingText.js
/*
|--------------------------------------------------------------------------
| Scrolling Text
|--------------------------------------------------------------------------
|
*/

function scrollingText(text) {
   let res = []
   for (let i = 0; i < text.length; i++) {
      res.push(text.toUpperCase())
      text = text.slice(1) + text.slice(0, 1);
   }
   return res;
}


function scrollingText(text) {
   text = text.toUpperCase();
   return [...text].map((_, i) => text.slice(i) + text.slice(0, i));
}


function scrollingText(text) {
   return Array.from({
      length: text.length
   }, (_, i) => (text.slice(i) + text.slice(0, i)).toUpperCase())
}


function scrollingText(text) {
   let res = []
   for (let i = 0; i < text.length; i++) {
      res.push((text.slice(i) + text.slice(0, i)).toUpperCase())
   }
   return res;
}


console.log(scrollingText("abc"), ["ABC", "BCA", "CAB"])

File: Python/7py 4/scrolling_text.py
"""
|--------------------------------------------------------------------------
| Scrolling Text
|--------------------------------------------------------------------------
|
"""

def scrolling_text(text):
   res = []
   for i in range(len(text)):
      res.append((text[i:] + text[:i]).upper())
   return res
   

def scrolling_text(text):
   return [(text[i:] + text[:i]).upper() for i in range(len(text))]


def scrolling_text(text):
   return list(map(lambda x: (text[x:]+text[:x]).upper(), range(len(text))))


def scrolling_text(text):
   return [ (text[i:]+text[:i]).upper() for i, e in enumerate(text)]


print(scrolling_text("abc"), ["ABC","BCA","CAB"])